colrev/ops/built_in/search_sources/eric.py

```diff
@@ -54,42 +76,183 @@ def validate_source(
+            return field_value
+    
+
+    
     @classmethod
     def add_endpoint(
         cls, search_operation: colrev.ops.search.Search, query: str
-    ) -> colrev.settings.SearchSource:
-        """Add SearchSource as an endpoint (based on query provided to colrev search -a )"""
-        raise NotImplementedError
+    ) -> typing.Optional[colrev.settings.SearchSource]:
+        """Add SearchSource as an endpoint (based on query provided to colrev search -a)"""
 
-    def run_search(
-        self, search_operation: colrev.ops.search.Search, rerun: bool
-    ) -> None:
-        """Run a search of ERIC"""
+        if "https://api.ies.ed.gov/eric/?" in query:
+            url_parsed = urllib.parse.urlparse(query)
+            query = urllib.parse.parse_qs(url_parsed.query)
+            search = query.get('search', [''])[0]
+            start = query.get('start', ['0'])[0]
+            rows = query.get('rows', ['2000'])[0]
+            if ':' in search:
+                search = ERICSearchSource.search_split(search)
+            filename = search_operation.get_unique_filename(
+                file_path_string=f"eric_{search}"
+            )
+            add_source = colrev.settings.SearchSource(
+                endpoint="colrev.eric",
+                filename=filename,
+                search_type=colrev.settings.SearchType.DB,
+                search_parameters={"query": search, 'start': start, 'rows': rows},
+                load_conversion_package_endpoint={"endpoint": "colrev.bibtex"},
+                comment="",
+            )
+            return add_source
+
+        return None
+
+    
+
+
+    def run_search(self, search_operation, rerun):
+        eric_feed = self.search_source.get_feed(
+            review_manager=search_operation.review_manager,
+            source_identifier=self.source_identifier,
+            update_only=(not rerun),
+        )
+        prev_record_dict_version = {}
+        full_url = self.build_search_url()
+    
+        response = requests.get(full_url)
+        if response.status_code == 200:
+            data = response.json()
+            records = search_operation.review_manager.dataset.load_records_dict()
+            
+            for doc in data['response']['docs']:
+                record_id = doc['id']
+                if record_id not in records:
+                    record_dict = self.create_record_dict(doc)
+                    updated_record_dict = self.update_record_fields(record_dict)
+                    record = colrev.record.Record(data=updated_record_dict)
+                    added = eric_feed.add_record(record=record)
+
+                    if added:
+                        search_operation.review_manager.logger.info(
+                            " retrieve " + record.data["ID"]
+                        )
+                        eric_feed.nr_added += 1
+                    else:
+                        changed = self.update_existing_record(
```

Comment:
Not sure what the rationale for extracting `update_existing_record(...)` into a separate method is. It could be called directly.